regparser/commands/outline_depths.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from regparser.tree.depth.derive import derive_depths
+
+import click
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument('markers', type=click.STRING, required=True)
+def outline_depths(markers) -> None:
+    """
+    Infer an outline's structure.
+    Return a list of outline depths for a given list of space-separated markers.
+    """
+
+    # Input is space-separated.
+    marker_list = markers.split(' ')
+    all_solutions = derive_depths(marker_list, [])
+    depths = {tuple(str(a.depth) for a in s) for s in all_solutions}.pop()
+
+    # Expected output is space-separated.
+    formatted_output = ' '.join(depths)
+
+    print(formatted_output)
```

Comment:
Note that we support Python 2.7, so we'll want to swap this with click's [`echo`](http://click.pocoo.org/5/utils/#printing-to-stdout) or import `print` from [`__future__`](https://docs.python.org/2/library/__future__.html)